lua/chatcommands/coinflip.lua

```diff
@@ -0,0 +1,153 @@
+CoinFlips = CoinFlips or {}
+
+RegisterChatCommand({'coin','coinflip'},function(ply,arg)
+	local t = string.Explode(" ",arg)
+	local p = tonumber(table.remove(t))
+	if #t == 1 and t[1]:lower() == "accept" then checkCoinFlipRequest(ply, p) return end
+
+	if p == nil then
+		if #t == 1 and t[1]:lower() == "accept" then
+			ply:ChatPrint("[orange]!coinflip accept points")
+		else
+			ply:ChatPrint("[orange]!coinflip player points")
+		end
+	else
+		local to,c = PlyCount(string.Implode(" ",t))
+		if c == 1 then
+			if ply == to then
+				ply:ChatPrint("[red]You can't coinflip yourself!")
+			elseif hasCoinflipRequest(to) then
+				ply:ChatPrint("[red]That player currently has a coinflip request active!")
+			elseif p >= 1000 then -- minimum 1000 point coinflip
+				initCoinFlip(ply,to,p)
+			else
+				ply:ChatPrint("[red]A coin flip must be a minimum of 1000 points. No pussy bets!")
+			end
+		else
+			ply:ChatPrint("[red]Player "..string.Implode(" ",t).." not found")
+		end
+	end
+end,{global=true,throttle=true})
+
+timer.Create( "CoinFlip", 1, 0,
+function()
+	local usedID = nil
+	for fromID,coinflip in pairs(CoinFlips) do
+		if((coinFlip[3] + 60) <= CurTime()) then -- If 60 seconds have elapsed
+			usedID = fromID -- Grab only 1 per timer itteration, prevents from having to table.remove inside of a loop.
+			break
+		end
+	end
+	if(usedID ~= nil) then
+		local coinflipData = CoinFlips[usedID]
+		local fromPlayer = player.GetBySteamID(usedID)
+		local toPlayer = player.GetBySteamID(coinflipData[1])
+		if(fromPlayer ~= nil and toPlayer ~= nil) then
+			fromPlayer:ChatPrint("[edgy]" .. toPlayer:Nick() .. "[fbc] don't want to play. Try again lates.")
+			toPlayer:ChatPrint("[fbc]You missed out on a coinflip from [edgy]" .. fromPlayer:Nick() .. "[fbc].")
+		elseif(fromPlayer == nil and toPlayer ~= nil) then
+			toPlayer:ChatPrint("[fbc]You missed out on a coinflip.")
+		elseif(fromPlayer ~= nil and toPlayer == nil) then
+			fromPlayer:ChatPrint("[fbc]The player you requested a coinflip to has left. Try again lates.")
+		end
+		table.remove(CoinFlips, usedID)
+	end
+end)
+
+function initCoinFlip(ply,target,amount)
+	if ply:PS_HasPoints(amount) and target:PS_HasPoints(amount) and CoinFlips[ply:SteamID()] == nil then
+		-- Both players have enough points AND an existing coinflip doesn't exist.
+		ply:ChatPrint("[orange]"..to:Nick().." is receiving your coinflip request.")
+		CoinFlips[ply:SteamID()] = {target:SteamID(), amount, CurTime()}
+	elseif CoinFlips[ply:SteamID()] ~= nil then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh already hae a filp goin!")
+	elseif not ply:PS_HasPoints(amount) then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh don't hae dooze points!")
+	elseif not target:PS_HasPoints(amount) then
+		-- Target doesn't have the required funds. Nothing happens, CoinFlipRequestTimeout will be called and timeout the roll.
```

Comment:
They can still infer from the "is recieving your coinflip request" thing. Just don't check the target's number of points but prohibit them from accepting it later if they don't have enough points.